src/complex/DataStructure/DataStore.hpp

```diff
@@ -371,11 +371,30 @@ class DataStore : public AbstractDataStore<T>
 
     // Create DataStore
     auto dataStore = std::make_unique<DataStore<T>>(tupleShape, componentShape, static_cast<T>(0));
-
     if(!datasetReader.readIntoSpan(dataStore->createSpan()))
     {
-      throw std::runtime_error(fmt::format("Error reading data from DataStore from HDF5 at {}/{}", H5::Support::GetObjectPath(datasetReader.getParentId()), datasetReader.getName()));
+      throw std::runtime_error(fmt::format("Error reading data array from DataStore from HDF5 at {}/{}", H5::Support::GetObjectPath(datasetReader.getParentId()), datasetReader.getName()));
+    }
+
+    return dataStore;
+  }
+
+  static std::unique_ptr<DataStore> ReadHdf5NeighborList(const H5::DatasetReader& datasetReader)
```

Comment:
I was thinking of moving this into the `NeighborList` files and directly read into a `std::vector`.

Reply:
There are some issues doing this as the code uses internal variables from the DataStore<T> class.